### ISSUE TITLE
cant build for msvc target in wsl

### ISSUE THREAD

**Issue by Hellzbellz123** [nabijaczleweli/rust-embed-resource#47]
getting some strange errors under wsl, its always looking for folder target/debug/build/**vanillacoffee-291d4c59a3e0df4c**/out/icon.lib but that vanillacoffee folder does exist but the icon-preprocessed.rc file is empty. placing  a populated file or icon.res output from llvm-rc does nothing and llvm-rc works fine. if someone knows whats up that would be helpful

```
❯ cargo build --target=x86_64-pc-windows-msvc
   Compiling vanillacoffee v0.0.1 (/home/hellzbellz/Source/VanillaCoffee)
The following warnings were emitted during compilation:

warning: cc: warning: icon.rc: linker input file unused because linking not done

error: failed to run custom build command for `vanillacoffee v0.0.1 (/home/hellzbellz/Source/VanillaCoffee)`

Caused by:
  process didn't exit successfully: `/home/hellzbellz/Source/VanillaCoffee/target/debug/build/vanillacoffee-7c8579190beb90b1/build-script-build` (exit status: 101)
  --- stdout
  running: "cc" "-O1" "-ffunction-sections" "-fdata-sections" "-g" "-fno-omit-frame-pointer" "-m64" "-I" "/home/hellzbellz/Source/VanillaCoffee/target/x86_64-pc-windows-msvc/debug/build/vanillacoffee-291d4c59a3e0df4c/out" "-Wall" "-Wextra" "-DRC_INVOKED" "-E" "icon.rc"
  cargo:warning=cc: warning: icon.rc: linker input file unused because linking not done
  exit status: 0

  --- stderr
  thread 'main' panicked at 'Couldn't execute llvm-rc to compile "/home/hellzbellz/Source/VanillaCoffee/target/x86_64-pc-windows-msvc/debug/build/vanillacoffee-291d4c59a3e0df4c/out/icon-preprocessed.rc" into "/home/hellzbellz/Source/VanillaCoffee/target/x86_64-pc-windows-msvc/debug/build/vanillacoffee-291d4c59a3e0df4c/out/icon.lib": No such file or directory (os error 2)', /home/hellzbellz/.cargo/registry/src/github.com-1ecc6299db9ec823/embed-resource-1.7.2/src/non_windows.rs:118:19
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

**Comment by nabijaczleweli** [comment]
Hm. Can you try using a checkout of `rust-embed-resource` with the following diff applied?
```diff
diff --git a/src/non_windows.rs b/src/non_windows.rs
index 5b9783e..c995f81 100644
--- a/src/non_windows.rs
+++ b/src/non_windows.rs
@@ -81,6 +81,7 @@ impl Compiler {
                 fs::write(&preprocessed_path,
                           cc::Build::new()
                               .define("RC_INVOKED", None)
+                              .flag("-xc")
                               .file(resource)
                               .cargo_metadata(false)
                               .include(out_dir)
```
It looks like your C compiler is misinterpreting the resource file as linker script.

**Comment by Hellzbellz123** [comment]
didnt get any changes with this i cloned and added flags manually cuz idk how to use difs 😣
```
error: failed to run custom build command for `vanillacoffee v0.0.1 (/home/hellzbellz/Source/VanillaCoffee)`

Caused by:
  process didn't exit successfully: `/home/hellzbellz/Source/VanillaCoffee/target/debug/build/vanillacoffee-ca59b0493012de7c/build-script-build` (exit status: 101)
  --- stdout
  running: "cc" "-O1" "-ffunction-sections" "-fdata-sections" "-g" "-fno-omit-frame-pointer" "-m64" "-I" "/home/hellzbellz/Source/VanillaCoffee/target/x86_64-pc-windows-msvc/debug/build/vanillacoffee-48973b06d6d14990/out" "-Wall" "-Wextra" "-xc" "-DRC_INVOKED" "-E" "icon.rc"
  exit status: 0

  --- stderr
  thread 'main' panicked at 'Couldn't execute llvm-rc to compile "/home/hellzbellz/Source/VanillaCoffee/target/x86_64-pc-windows-msvc/debug/build/vanillacoffee-48973b06d6d14990/out/icon-preprocessed.rc" into "/home/hellzbellz/Source/VanillaCoffee/target/x86_64-pc-windows-msvc/debug/build/vanillacoffee-48973b06d6d14990/out/icon.lib": No such file or directory (os error 2)', crates/rust-embed-resource/src/non_windows.rs:119:19
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

**Comment by nabijaczleweli** [comment]
uuuh, do you actually have `llvm-rc`?

**Comment by Hellzbellz123** [comment]
output of llvm-rc is below so i assume i have it
❯ llvm-rc icon.rc /v
 /usr/sbin/clang --driver-mode=gcc -target x86_64-pc-windows-msvc-coff -E -xc -DRC_INVOKED icon.rc -o /tmp/preproc-875e8b.rc
Identifier: app_icon
Identifier: ICON
String: "test.ico"
Icon (app_icon): "test.ico"

**Comment by nabijaczleweli** [comment]
Hm, yeah, that's odd. If `/home/hellzbellz/Source/VanillaCoffee/target/x86_64-pc-windows-msvc/debug/build/vanillacoffee-48973b06d6d14990/out` exists (and it must since we just wrote `icon-preprocessed.rc` under it) then I don't really see how this could be possible. Can you post the result of `strace -fo trace -s9999 cargo {build/run/whatever arguments you're using}`?

**Comment by Hellzbellz123** [comment]
idek where to post a 280mb file

**Comment by Hellzbellz123** [comment]
[trace.log](https://github.com/nabijaczleweli/rust-embed-resource/files/9033819/trace.log)

**Comment by nabijaczleweli** [comment]
hm, the only thing that stands out here is
```
16926 chdir("" <unfinished ...>
16926 <... chdir resumed>)              = -1 ENOENT (No such file or directory)
```

What do you get with this diff?
```
diff --git a/src/non_windows.rs b/src/non_windows.rs
index c995f81..ba34c0f 100644
--- a/src/non_windows.rs
+++ b/src/non_windows.rs
@@ -91,7 +91,7 @@ impl Compiler {
                 try_command(Command::new(&self.executable[..])
                                 .args(&["/fo", &out_file, "--", &preprocessed_path])
                                 .stdin(Stdio::piped())
-                                .current_dir(Path::new(resource).parent().expect("Resource parent nonexistent?")),
+                                .current_dir(dbg!(Path::new(dbg!(resource)).parent()).expect("Resource parent nonexistent?")),
                             Path::new(&self.executable[..]),
                             "compile",
                             &preprocessed_path,
```

**Comment by Hellzbellz123** [comment]
this was run in a new wsl instance

```
   Compiling vanillacoffee v0.0.1 (/home/hellzbellz/Source/VanillaCoffee)
   Compiling tiff v0.7.2
error: failed to run custom build command for `vanillacoffee v0.0.1 (/home/hellzbellz/Source/VanillaCoffee)`

Caused by:
  process didn't exit successfully: `/home/hellzbellz/Source/VanillaCoffee/target/debug/build/vanillacoffee-ca59b0493012de7c/build-script-build` (exit status: 101)
  --- stdout
  running: "cc" "-O1" "-ffunction-sections" "-fdata-sections" "-g" "-fno-omit-frame-pointer" "-m64" "-I" "/home/hellzbellz/Source/VanillaCoffee/target/x86_64-pc-windows-msvc/debug/build/vanillacoffee-00f2cfe475c8bc67/out" "-Wall" "-Wextra" "-xc" "-DRC_INVOKED" "-E" "icon.rc"
  exit status: 0

  --- stderr
  [crates/rust-embed-resource/src/non_windows.rs:94] resource = "icon.rc"
  [crates/rust-embed-resource/src/non_windows.rs:94] Path::new(dbg!(resource)).parent() = Some(
      "",
  )
  thread 'main' panicked at 'Couldn't execute llvm-rc to compile "/home/hellzbellz/Source/VanillaCoffee/target/x86_64-pc-windows-msvc/debug/build/vanillacoffee-00f2cfe475c8bc67/out/icon-preprocessed.rc" into "/home/hellzbellz/Source/VanillaCoffee/target/x86_64-pc-windows-msvc/debug/build/vanillacoffee-00f2cfe475c8bc67/out/icon.lib": No such file or directory (os error 2)', crates/rust-embed-resource/src/non_windows.rs:119:19
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
[hellzbellz@BertiebottPRIME VanillaCoffee]$ 

```

**Comment by nabijaczleweli** [comment]
Ha! Can you try the current `master` branch (at least 1eb89effba9ad209df352fdb4dd141ab2fe9352f)?

**Comment by Hellzbellz123** [comment]
ooopsies lol yes recloning

**Comment by Hellzbellz123** [comment]
most recent commit fixed the issue for me :) judging by commit issues missing slashes somewhere but where was it my config?

**Comment by nabijaczleweli** [comment]
Nah, it was a fucky wucky on my part (well, I kinda expected the parent of `"file"` to be `"."`, not `""`, but).

Fixed in [`v1.7.3`](https://github.com/nabijaczleweli/rust-embed-resource/releases/tag/v1.7.3), backported to [`v1.6.9`](https://github.com/nabijaczleweli/rust-embed-resource/releases/tag/v1.6.9).
